Framework/DataHandling/src/LoadSESANS.cpp

```diff
@@ -0,0 +1,419 @@
+  auto j = find_if(i, line.end(), &space);
+
+  if (j == line.end())
+    throwFormatError(line, "Expected key-value pair", lineNum);
+
+  attribute.first = std::string(i, j);
+
+  // Find start of second word
+  i = find_if(j, line.end(), notSpace);
+  if (i == line.end())
+    throwFormatError(line, "Expected key-value pair", lineNum);
+
+  // Grab from there to the end of the line
+  attribute.second = std::string(i, line.end());
+
+  return attribute;
+}
+
+/** Helper function to throw an error relating to the format of the file
+* @param line The line where a format error was found
+* @param message A message detailing the reason for the error
+* @param lineNum The number of the line
+* @throw runtime_error Under all circumstances
+*/
+void LoadSESANS::throwFormatError(const std::string &line,
+                                  const std::string &message,
+                                  const int &lineNum) {
+  std::string output = "Badly formed line at line " + std::to_string(lineNum) +
+                       ": \"" + line + "\"\n(" + message + ")";
+  g_log.error(output);
+  throw std::runtime_error(output);
+}
+
+/** Make sure that all mandatory headers are supplied in the file
+* @throw runtime_error If any other the mandatory headers are missing
+*/
+void LoadSESANS::checkMandatoryHeaders() {
+  for (std::string attr : mandatoryAttributes) {
+    // if (std::find(attributes.begin(), attributes.end(), attr) ==
+    // attributes.end()){
+    // if (attributes.find(attr) == attributes.end()) {
+    if (!attributes.count(attr)) {
+      std::string err = "Failed to supply parameter: \"" + attr + "\"";
+      g_log.error(err);
+      throw std::runtime_error(err);
+    }
+  }
+}
+
+/** Create a new workspace with the columns read from the file
+ * @param columns Data columns from input file
+ * @return A workspace with the corresponding data
+ */
+API::MatrixWorkspace_sptr LoadSESANS::makeWorkspace(ColumnMap columns) {
+  size_t histogramLength = columns["SpinEchoLength"].size();
+  API::MatrixWorkspace_sptr newWorkspace =
+      API::WorkspaceFactory::Instance().create(
+          "Workspace2D", 1, histogramLength, histogramLength);
+
+  auto xValues = columns["Wavelength"];
+  auto yValues =
+      calculateYValues(columns["Depolarisation"], columns["Wavelength"]);
+  auto eValues = calculateEValues(columns["Depolarisation_error"], yValues,
+                                  columns["Wavelength"]);
+
+  auto &dataX = newWorkspace->mutableX(0);
+  auto &dataY = newWorkspace->mutableY(0);
+  auto &dataE = newWorkspace->mutableE(0);
+
+  for (size_t i = 0; i < histogramLength; i++) {
+    dataX[i] = xValues[i];
+    dataY[i] = yValues[i];
+    dataE[i] = eValues[i];
+  }
+
+  return newWorkspace;
+}
+
+/** Is a character whitespace (here considered space or tab)?
+ * @param c The character
+ * @return Whether it is whitespace
+ */
+bool LoadSESANS::space(const char &c) { return c == ' ' || c == '\t'; }
```

Comment:
This is not really required here, but just for your info. These helper functions could live in a anonymous namespace in the `cpp` file. This also makes them `invisible` to other modules, but ensures that when you change them you will not have to recompile elements which depend on `SaveSESANS`.